pkg/manager/modelregistry.go

```diff
@@ -79,6 +91,69 @@ func (m *Manager) Capabilities() []*gnmi.ModelData {
 	return outputList
 }
 
-func toModelName(name string, version string) string {
+// extractReadOnlyPaths is a recursive function to extract a list of read only paths from a YGOT schema
+func extractReadOnlyPaths(deviceEntry *yang.Entry, parentState yang.TriState, parentPrefix string, parentPath string) []string {
+	readOnlyPaths := make([]string, 0)
+
+	for _, dirEntry := range deviceEntry.Dir {
+		var itemPath string
+		if parentPath == "" {
+			itemPath = fmt.Sprintf("%s/%s:%s", parentPath, dirEntry.Prefix.Name, dirEntry.Name)
+		} else {
+			itemPath = fmt.Sprintf("%s/%s", parentPath, dirEntry.Name)
+		}
+		//fmt.Println("Processing", itemPath)
+		if dirEntry.IsLeaf() {
+			// No need to recurse
+			if dirEntry.Config == yang.TSFalse || parentState == yang.TSFalse {
+				readOnlyPaths = append(readOnlyPaths, itemPath)
+			}
+		} else if dirEntry.IsContainer() {
+			var name string
+			if dirEntry.Prefix.Name == parentPrefix {
+				name = parentPath + "/" + dirEntry.Name
+			} else {
+				name = parentPath + "/" + dirEntry.Prefix.Name + ":" + dirEntry.Name
+			}
+			if dirEntry.Config == yang.TSFalse || parentState == yang.TSFalse {
+				readOnlyPaths = append(readOnlyPaths, itemPath)
+			}
+			newPaths := extractReadOnlyPaths(dirEntry, dirEntry.Config, dirEntry.Prefix.Name, name)
+			for _, newPath := range newPaths {
+				readOnlyPaths = append(readOnlyPaths, newPath)
+			}
+		} else if dirEntry.IsList() {
+			var name string
+			if dirEntry.Prefix.Name == parentPrefix {
+				name = parentPath + "/" + dirEntry.Name + "[" + dirEntry.Key + "='*']"
```

Comment:
can we use fmt.Sprintf as before. I'd actually make a function out of it for simplicity reason and also for avoiding to forget formatting changes here and there.

Reply:
all functioned up now ;-)